frontend-react/src/components/StaticCompare.tsx

```diff
@@ -1,150 +1,91 @@
-            // initialization only
-            refreshHighlights(props.leftText, props.rightText);
-        }
-    }, [props.leftText, props.rightText, refreshHighlights]);
+        refreshHighlights(props.leftText, props.rightText, props.jsonDiffMode);
+    }, [
+        props.leftText,
+        props.rightText,
+        props.jsonDiffMode,
+        refreshHighlights,
+    ]);
 
     return (
-        <ScrollSync>
-            <div className="rs-editable-compare-container">
-                {/*Text has two components. The text on top and the highlight under it.*/}
-                {/*scrolling is synced so they line up*/}
-
-                <div className="rs-editable-stacked-container">
-                    <ScrollSyncPane>
-                        <div
-                            className="rs-static-compare-base rs-editable-compare-static"
-                            data-testid={"left-compare-text"}
-                            contentEditable={false}
-                            dangerouslySetInnerHTML={{
-                                __html: DOMPurify.sanitize(props.leftText),
-                            }}
-                        />
-                    </ScrollSyncPane>
-                    <ScrollSyncPane>
-                        <div
-                            className="rs-static-compare-base rs-editable-compare-background"
-                            data-testid={"left-compare-highlight"}
-                            dangerouslySetInnerHTML={{
-                                __html: DOMPurify.sanitize(leftHighlightHtml),
-                            }}
-                        />
-                    </ScrollSyncPane>
-                </div>
-
-                {/*Text has two components. The text on top and the highlight under it.*/}
-                {/*scrolling is synced so they line up*/}
-                <div className="rs-editable-stacked-container">
-                    <ScrollSyncPane>
-                        <div
-                            className="rs-static-compare-base rs-editable-compare-static"
-                            data-testid={"right-compare-text"}
-                            dangerouslySetInnerHTML={{
-                                __html: DOMPurify.sanitize(props.rightText),
-                            }}
-                        />
-                    </ScrollSyncPane>
-
-                    <ScrollSyncPane>
-                        <div
-                            className="rs-static-compare-base rs-editable-compare-background"
-                            data-testid={"right-compare-highlight"}
-                            dangerouslySetInnerHTML={{
-                                __html: `${DOMPurify.sanitize(
-                                    rightHighlightHtml
-                                )}<br/>`,
-                            }}
-                        />
-                    </ScrollSyncPane>
+        <>
```

Comment:
is this change just related to wrapping in a fragment in order to add the checkbox?

Reply:
Yes. I was trying to do a demo and couldn't line up the left and right sides and it was frustrating, so this is an experiment to see if disabling the scroll sync is useful.